src/backgroundGCP/DataIngestionGCP/Functions/GetCopilotSeatsByDate.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+using Google.Cloud.Firestore;
+using Google.Cloud.Functions.Framework;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.CopilotDashboard.DataIngestion.Functions;
+
+public class GetCopilotSeatsByDate : IHttpFunction
+{
+    private readonly ILogger<GetCopilotSeatsByDate> _logger;
+    private readonly FirestoreDb _firestoreDb;
+    private readonly JsonSerializerOptions jsonSerializerOptions = new()
+    {
+        Converters = { new FirestoreTimestampConverter() }
+    };
+
+    public GetCopilotSeatsByDate(
+    ILogger<GetCopilotSeatsByDate> logger,
+    FirestoreDb firestoreDb)
+    {
+        _logger = logger;
+        _firestoreDb = firestoreDb;
+    }
+
+    public async Task HandleAsync(HttpContext context)
+    {
+        try
+        {
+            string apiKey = context.Request.Query["apiKey"];
+            if (string.IsNullOrEmpty(apiKey))
+            {
+                context.Response.StatusCode = StatusCodes.Status400BadRequest;
+                await context.Response.WriteAsync("Missing 'apiKey' parameter.");
+                return;
+            }
+
+            if (apiKey != Environment.GetEnvironmentVariable("API_KEY"))
+            {
+                context.Response.StatusCode = StatusCodes.Status401Unauthorized;
+                await context.Response.WriteAsync("Invalid API key.");
+                return;
+            }
+
+            string fromDateStr = context.Request.Query["from"];
+            string toDateStr = context.Request.Query["to"];
+
+            DateTime? fromDate = null;
+            DateTime? toDate = null;
+
+            if (!string.IsNullOrEmpty(fromDateStr))
+            {
+                if (!DateTime.TryParse(fromDateStr, out DateTime parsedFromDate))
+                {
+                    context.Response.StatusCode = StatusCodes.Status400BadRequest;
+                    await context.Response.WriteAsync("Invalid 'from' date.");
+                    return;
+                }
+                fromDate = parsedFromDate;
+            }
+
+            if (!string.IsNullOrEmpty(toDateStr))
+            {
+                if (!DateTime.TryParse(toDateStr, out DateTime parsedToDate))
```

Comment:
I think better to specify the format otherwise I believe it will take the format day according to the server's date settings

Reply:
Added format